aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -85,6 +88,30 @@ internal class AWSS3StorageService(
         return URL(presignUrlRequest.url.toString())
     }
 
+    /**
+     * Validate if S3 object exists for the given key.
+     * Throws StorageException if NoSuchKey S3 client exception is caught.
+     * @param serviceKey S3 service key
+     */
+    override fun validateObjectExists(serviceKey: String) {
+        try {
+            runBlocking {
+                s3Client.headObject(
+                    HeadObjectRequest {
+                        bucket = s3BucketName
+                        key = serviceKey
+                    }
+                )
+            }
+        } catch (ex: NotFound) {
```

Comment:
I'm guessing this is the same implementaton as Swift? to check head object and catch NotFound. Odd the documentation shows "NoSuchKey" error and nothing about "NotFound" https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#API_HeadObject_Errors

Reply:
yes, this exception type is the same on [Amplify Swift](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService%2BGetPreSignedURLBehavior.swift#L35C26-L35C34) and verified with E2E testing.